src/Frontend/Modules/MailMotor/Actions/Subscribe.php

```diff
@@ -0,0 +1,252 @@
+<?php
+
+namespace Frontend\Modules\MailMotor\Actions;
+
+/*
+ * This file is part of the Fork CMS MailMotor Module from SIESQO.
+ *
+ * For the full copyright and license information, please view the license
+ * file that was distributed with this source code.
+ */
+
+use Frontend\Core\Engine\Base\Block as FrontendBaseBlock;
+use Frontend\Core\Engine\Language;
+use Frontend\Core\Engine\Form as FrontendForm;
+use Frontend\Core\Engine\Navigation as FrontendNavigation;
+use Frontend\Modules\MailMotor\Engine\Model as FrontendMailMotorModel;
+use MailMotor\Bundle\MailMotorBundle\Exception\NotImplementedException;
+
+/**
+ * This is the Subscribe-action for our MailMotor
+ *
+ * @author Jeroen Desloovere <jeroen@siesqo.be>
+ */
+class Subscribe extends FrontendBaseBlock
+{
+    /**
+     * FrontendForm instance
+     *
+     * @var	FrontendForm
+     */
+    private $frm;
+
+    /**
+     * @var array
+     */
+    private $interests;
+
+    /**
+     * Execute the extra
+     *
+     * @return void
+     */
+    public function execute()
+    {
+        parent::execute();
+        $this->loadData();
+        $this->loadTemplate();
+        $this->loadForm();
+        $this->validateForm();
+        $this->parse();
+    }
+
+    /**
+     * Load the form
+     *
+     * @return void
+     */
+    private function loadData()
+    {
+        try {
+            // Define interests
+            $this->interests = $this->get('mailmotor.subscriber')->getInterests();
+        // Fallback for when no mail-engine is chosen in the Backend
+        } catch (NotImplementedException $e) {
+            $this->interests = false;
+        }
+    }
+
+    /**
+     * Load the form
+     *
+     * @return void
+     */
+    private function loadForm()
+    {
+        // create the form
+        $this->frm = new FrontendForm('mailMotorSubscribeForm');
+
+        // define email
+        $email = null;
+
+        // request contains an email
+        if ($this->get('request')->request->get('email') != null) {
+            $email = $this->get('request')->request->get('email');
+        }
+
+        // create & add elements
+        $this->frm->addText('email', $email);
+        $this->frm->addText('email')
+            ->setAttributes(
+                array(
+                    'required' => null,
+                    'type' => 'email',
+                    'placeholder' => ucfirst(Language::lbl('YourEmail')),
+                )
+            )
+        ;
+
+        // Has interests
+        if ($this->interests) {
```

Comment:
you should check here if the intrests array isn't empty instead of mixing types with a bool